### ISSUE TITLE
low accuracy when batch_size = 1 or 2

### ISSUE THREAD

**Issue by lzhuwen** [charlesq34/pointnet#37]
HI Charles,
Thank you for this great work. 
I found something very strange: in the classification task (train.py), the training and testing accuracy are very low (around 0.1) when i set the batch size to 1 or 2. Do you have any idea why this happens?

I want to use batch size 1 because I want to process point clouds with varying number of points.

Best,

**Comment by daerduoCarey** [comment]
Hi, @lzhuwen ,

Thank you for your question. First of all, you should definitely avoid using such small batch size, because of many reasons. 1) We are using batch normalization layer during training. It's also good to have batch size no smaller than 32 I guess. 2) Batch size 1 or 2 is too small so that the gradient may change abruptly. This may cause your training longer. But, I guess probably in your case, you suffer from the problem because of reason 1).

For your second question, as we stated in the paper, you can always make the number of points the same for every training batches by repeating the first/last point several times. This will not affect the training if they are of similar numbers of points. During testing, you don't need to make them have the same number of points at all.

Hope this answers your question. Thank you!

Bests,
Kaichun

**Comment by lzhuwen** [comment]
Very helpful! Thanks Kaichun.

Best

**Comment by xinyugaotudelft** [comment]
Hi, @daerduoCarey ,

What if I remove the BN and make it batch size 1? Is this against the intention of the network design?

Best,
